csharp/src/Apache.Arrow/Arrays/StructArray.cs

```diff
@@ -21,13 +21,13 @@ namespace Apache.Arrow
 {
     public class StructArray : Array
     {
-        private readonly List<Array> _fields;
+        private readonly IEnumerable<IArrowArray> _fields;
```

Comment:
1. I agree. I had `Array` first, but I was forced to make it `IArrowArray` because `RecordBatch.Column(i)` returns an `IArrowArray`. See https://github.com/dotnet/spark/blob/7bcd2a5060e12ca993132f4986b2dfb4b07df94f/src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs#L365. When we refactor `IArrowArray` -> `ArrowArray`, this can change too.
2. Yeap, you're right. Removed it. It is unused at the moment.

Reply:
I now fill out the `_fields` data correctly.